metrics/web/docs/features/deep_links/diagrams/deep_links_integration_component_diagram.puml

```diff
@@ -0,0 +1,35 @@
+@startuml deep_links_integration_component_diagram
+
+skinparam componentStyle rectangle
+
+component Browser #A9DCDF
+component RouteInformationProvider
+component RouteInformationParser
+component RouterDelegate
+component NavigationState
+component NavigationNotifier
+component ApplicationState
+
+Browser -down-> RouteInformationProvider : URL
+
+RouteInformationProvider -down-> RouteInformationParser : routeInformation
+
+RouteInformationParser -down-> RouterDelegate : routeConfiguration
+
+RouterDelegate -down-> NavigationNotifier : delegate handling\n new routeConfiguration
+
+NavigationNotifier -left-> NavigationState : updates
+
+NavigationNotifier --> ApplicationState : updates
+
+ApplicationState --> NavigationNotifier : UI events \n (e.g. filtering, searching, \n back button pressed)
+
+NavigationNotifier -up-> RouterDelegate : notifies about \n navigation state updates
```

Comment:
It seems like the `RouterDelegate` and other` Navigator 2.0` components do not participate in the process of changing the link according to changed filters/dropdown selections/etc.